File: app/src/main/java/com/arturogutierrez/openticator/domain/password/wizard/di/MasterPasswordModule.java
package com.arturogutierrez.openticator.domain.password.wizard.di;

import com.arturogutierrez.openticator.di.PerActivity;
import com.arturogutierrez.openticator.domain.Preferences;
import com.arturogutierrez.openticator.domain.password.PasswordSerializer;
import com.arturogutierrez.openticator.domain.password.PasswordSerializerImpl;
import com.arturogutierrez.openticator.domain.password.interactor.CreateMasterPasswordInteractor;
import dagger.Module;
import dagger.Provides;

@Module
public class MasterPasswordModule {

  public MasterPasswordModule() {

  }

  @Provides
  PasswordSerializer providePasswordSerializer(PasswordSerializerImpl passwordSerializer) {
    return passwordSerializer;
  }

  @Provides
  @PerActivity
  CreateMasterPasswordInteractor provideCreateMasterPasswordInteractor(Preferences preferences,
      PasswordSerializer passwordSerializer) {
    return new CreateMasterPasswordInteractor(preferences, passwordSerializer);
  }
}

File: app/src/main/kotlin/com/arturogutierrez/openticator/domain/password/wizard/di/MasterPasswordModule.kt
package com.arturogutierrez.openticator.domain.password.wizard.di

import com.arturogutierrez.openticator.di.PerActivity
import com.arturogutierrez.openticator.domain.password.PasswordSerializer
import com.arturogutierrez.openticator.domain.password.PasswordSerializerImpl
import com.arturogutierrez.openticator.domain.password.interactor.CreateMasterPasswordInteractor
import com.arturogutierrez.openticator.storage.preferences.Preferences
import dagger.Module
import dagger.Provides

@Module
class MasterPasswordModule {

  @Provides
  fun providePasswordSerializer(): PasswordSerializer = PasswordSerializerImpl()

  @Provides
  @PerActivity
  fun provideCreateMasterPasswordInteractor(preferences: Preferences,
                                            passwordSerializer: PasswordSerializer): CreateMasterPasswordInteractor {
    return CreateMasterPasswordInteractor(preferences, passwordSerializer)
  }
}
